qa/sc_import_export_keys.py

```diff
@@ -0,0 +1,252 @@
+from SidechainTestFramework.sc_test_framework import SidechainTestFramework
+from test_framework.util import assert_equal, assert_false, assert_true, initialize_chain_clean, start_nodes, websocket_port_by_mc_node_index, forward_transfer_to_sidechain
+from SidechainTestFramework.scutil import start_sc_nodes, generate_next_blocks, bootstrap_sidechain_nodes, connect_sc_nodes
+from httpCalls.wallet.createPrivateKey25519 import http_wallet_createPrivateKey25519
+from SidechainTestFramework.sc_boostrap_info import SCNodeConfiguration, SCCreationInfo, MCConnectionInfo, \
+    SCNetworkConfiguration
+from httpCalls.wallet.exportSecret import http_wallet_exportSecret
+from httpCalls.wallet.allPublicKeys import http_wallet_allPublicKeys
+from httpCalls.wallet.importSecret import http_wallet_importSecret
+from httpCalls.wallet.dumpSecrets import http_wallet_dumpSecrets
+from httpCalls.wallet.importSecrets import http_wallet_importSecrets
+from httpCalls.wallet.balance import http_wallet_balance
+from httpCalls.transaction.sendCoinsToAddress import sendCoinsToAddress
+from SidechainTestFramework.sidechainauthproxy import SCAPIException
+
+"""
+    - Setup 2 SC Node.
+    - Create a new address on node2
+    - Export its private key and import it inside the node1
+    - Verify that node1 owns this address
+    - Send coins to this address and verify that the balance on node1 is unchanged and node2 owns the new amount
+    
+    - Create some new addresses on node1 and dump all its secret on file
+    - Import the node1 secrets inside the node2 and verify that we imported only the secrets that were missing
+"""
+class SidechainImportExportKeysTest(SidechainTestFramework):
+    number_of_mc_nodes = 1
+    number_of_sidechain_nodes = 2
+    withdrawalEpochLength=10
+    API_KEY = "Horizen"
+    API_KEY_HASH = "aa8ed2a907753a4a7c66f2aa1d48a0a74d4fde9a6ef34bae96a86dcd7800af98"
+
+    def setup_chain(self):
+        initialize_chain_clean(self.options.tmpdir, self.number_of_mc_nodes)
+
+    def setup_network(self, split = False):
+        # Setup nodes
+        self.nodes = self.setup_nodes()
+        self.sync_all()
+
+    def setup_nodes(self):
+        # Start 1 MC nodes
+        return start_nodes(self.number_of_mc_nodes, self.options.tmpdir)
+
+    def sc_setup_chain(self):
+        # Bootstrap new SC, specify SC node 1 and SC node 2 connection to MC node 1
+        mc_node_1 = self.nodes[0]
+
+        sc_node_1_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_1.hostname, websocket_port_by_mc_node_index(0))),
+            True,
+            automatic_fee_computation=False,
+            api_key_hash=self.API_KEY_HASH
+        )
+        sc_node_2_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_1.hostname, websocket_port_by_mc_node_index(0))),
+            True,
+            automatic_fee_computation=False,
+            api_key_hash=self.API_KEY_HASH
```

Comment:
API key and key hash can be generated automatically during node bootstrap. Use bootstrapping tool new command.

Reply:
Done